### ISSUE TITLE
Angular 10 Router Outlet does not match

### ISSUE THREAD

**Issue by danielehrhardt** [ionic-team/ionic#21661]
# 🐞 bug report

### Affected Package

 The issue is caused by package @angular/....
@angular/*

### Is this a regression?
Yes
I downgraded my Version to ~9.1.8 and there it is working.

### Description
![image](https://user-images.githubusercontent.com/19570043/86025880-0b0f4200-ba2f-11ea-9223-80ab0d718414.png)

this.router.navigate(['auth', 'login']) should redirect to the Router Outlet inside the Blank Component
this.router.navigate(['dashboard']) should redirect to the Router Outlet inside the Full Component

With Angular 10 the Outlet stay in the Old URL Outlet.

https://github.com/codextde/ionic-router-angular-10-issue

Angular Issue:
https://github.com/angular/angular/issues/37826

## 🌍  Your Environment

**Angular Version:**
Angular 10

**Comment by liamdebeasi** [comment]
Thanks for the issue. Can you please provide some steps to reproduce? I am not seeing the issue you are describing.

**Comment by danielehrhardt** [comment]
I created a sample Repo.
https://github.com/codextde/ionic-router-angular-10-issue
Klick on the Button and then you will see the Issue

**Comment by liamdebeasi** [comment]
Yes, I saw the repo you provided. What I need is some steps to reproduce the issue so I can verify whether or not this is a bug in Ionic Framework.

**Comment by danielehrhardt** [comment]
Ionic Router:
`<ion-router-outlet></ion-router-outlet>
`
![screen-capture](https://user-images.githubusercontent.com/19570043/86367662-89145880-bc7c-11ea-82b9-a30c0ec7bf5a.gif)


Angular Router
`<router-outlet></router-outlet>`
![screen-capture-_1_](https://user-images.githubusercontent.com/19570043/86367760-a517fa00-bc7c-11ea-86fe-3c1c6d5f58d9.gif)

**Comment by liamdebeasi** [comment]
Can you please describe the expected behavior using words rather than 2 images? It is hard to parse what is going on in the GIFs.

**Comment by danielehrhardt** [comment]
1. You need to create two Components in my Example Full and Blank. (In my case the Blank Component represent the wrapper for Pages like the Login, and the Full Component for Stuff with a Sidemenu (Admin Panel))
2. Then you need to create two Pages with Lazy Loading 
3. Then you need to adjust the Router. That the first Page is inside the Blank Component and the second Page goes inside the Full Component
4. When you then add a Button in the First Page with a redirect to the Second Page it should switch from the Blank Wrapper to the Full Wrapper

The Ionic Route behavior is that the Second Page stays in the Blank Wrapper.

**Comment by mhartington** [comment]
This sort of router setup probably should have never worked. 
Aside from the developer UX of having multiple path matches in the top level, which can be confusing, having multiple nested router outlets has not been something we supported in any version of Ionic.

**Comment by danielehrhardt** [comment]
Could then please provide an alternative?
If we look at the Ionic Side Menu Sample App there we have the Side Menu in the Root Level. 
![image](https://user-images.githubusercontent.com/19570043/86378569-17dba200-bc8a-11ea-9ab5-213d3c0db63b.png)
How would you disable it there?

**Comment by liamdebeasi** [comment]
It might be easier to disable/enable the side menu based upon the login state of the application: https://ionicframework.com/docs/api/menu#properties.

I am going to close this as the original application setup is not something we support, as Mike mentioned. Thanks!

**Comment by danielehrhardt** [comment]
hm, this is a bit frustrating. In Angular, it is also working and before Version 10 it also worked with Ionic. But okay. Thank you for the fast support.

**Comment by NikolaPeevski** [comment]
@danielehrhardt In the app component subscribe to the router with an instance on navigation end, then just toggle the menu/ngIf it based on the path, it's quite simple to have conditionals based on routes.

**Comment by danielehrhardt** [comment]
> @danielehrhardt In the app component subscribe to the router with an instance on navigation end, then just toggle the menu/ngIf it based on the path.

I did this before but with your solution you are really unflexible. 

I mean with my option you can create wrapper for everything. Not only the side Menu. 

We use ionic as a additional layer in our Angular Environment. 

In my example it was only a small area to show the issue. 


Why we should put stuff in a lazy loading HTML module that is not necessary. 
Why we should let the Frontend waste rendering power to parse the ngif? There is really no need for it.

**Comment by NikolaPeevski** [comment]
Well that was just a top level example, you can do it however you'd want. Ionic is a tool in the end, like every other framework, language etc. It can't be tailor made specifically for you. I agree with @mhartington that paths should be unique, especially if you factor the possibility of automated rerouting using guards or some other sort, like partial path matching with a hierarchy manner. You're not really obligated to lazy load everything. There are different loading strategies for your app. You can use additional router outlets as wrappers with different ids, possibilities are endless. Lastly if you're concerned about browser rendering performance look at SSR. :)

**Comment by danielehrhardt** [comment]
> This sort of router setup probably should have never worked. 
> Aside from the developer UX of having multiple path matches in the top level, which can be confusing, having multiple nested router outlets has not been something we supported in any version of Ionic.
> 

There is one Question what I am asking myself. 
If we imagine we have tabs (not the ionic tabs more like generic Tabs (like the segment)) and want to lazy load the content inside the Tabs. 

We need to have nested Router Outlets. 

What is the reason to restrict it to only one layer?

**Comment by danielehrhardt** [comment]
> Well that was just a top level example, you can do it however you'd want. Ionic is a tool in the end, like every other framework, language etc. It can't be tailor made specifically for you. I agree with @mhartington that paths should be unique, especially if you factor the possibility of automated rerouting using guards or some other sort, like partial path matching with a hierarchy manner. You're not really obligated to lazy load everything. There are different loading strategies for your app. You can use additional router outlets as wrappers with different ids, possibilities are endless. Lastly if you're concerned about browser rendering performance look at SSR. :)

Thank you for the Reply :)
